clients/src/main/java/org/apache/kafka/common/record/CompressionType.java

```diff
@@ -96,9 +117,9 @@ public OutputStream wrapForOutput(ByteBufferOutputStream buffer, byte messageVer
         }
 
         @Override
-        public InputStream wrapForInput(ByteBufferInputStream buffer, byte messageVersion) {
+        public InputStream wrapForInput(ByteBuffer buffer, byte messageVersion) {
```

Comment:
For consistency, it may make sense for `wrapForOutput` to take a `ByteBuffer` too.